File: array/findpair.py
def find_pair(arr: list[int], target: int) -> None:

    n = len(arr)

    found = False

    for i in range(n - 1):

        for j in range(i + 1, n):

            if arr[i] + arr[j] == target:

                print(f"Pair found ({arr[i]}, {arr[j]})")

                found = True

    if not found:
        print("Pair not found")


nums = [8, 7, 2, 5, 3, 1]
target = 10

find_pair(nums, target)

File: array/findpair.go
package main


import "fmt"

// Given an unsorted integer array,
// find all pairs whose sum equals the target value.
//
// Example:
// nums = [8, 7, 2, 5, 3, 1]
// target = 10
//
// Output:
// (8,2)
// (7,3) 
// O(n^2)
func findPair(nums []int,target int){
	n := len(nums)-1

	found := false

	for i:=0;i<n-1;i++{
		for j:=i+1;j<n;j++{
		
			if nums[i] + nums[j] == target{
				
				fmt.Printf("Pair found (%v,%v)\n",nums[i],nums[j])
				
				found = true	
			}
		}
	}
		if !found{
					fmt.Println("Pair not found")
		}

}

func main(){

	nums := []int{8,7,2,5,3,1}

	findPair(nums,10)
}
